### PR TITLE
REF: remove BlockManager.set

### PR DESCRIPTION
**Pull request by jbrockmendel**
xref #33052, #33332.

This touches several DataFrame/NDFrame methods.  Those methods need a refactor in part because the ones defined on NDFrame are only ever called for DataFrame.  Will refactor in follow-up.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3210,7 +3210,7 @@ def _maybe_cache_changed(self, item, value) -> None:
         """
         The object has called back to us saying maybe it has changed.
         """
-        self._mgr.set(item, value)
+        NDFrame._set_item(self, item, value, clear=False)
```
Comment (simonjayhawkins) [inline pandas/core/generic.py:3213]: for my edification, why NDFrame?

Comment (jbrockmendel) [reply pandas/core/generic.py:3213]: because we dont want to go through the DataFrame._set_item method

Comment (simonjayhawkins) [reply pandas/core/generic.py:3213]: are double underscore methods appropriate here for intraclass method calls? https://docs.python.org/3/tutorial/classes.html?highlight=mangle#private-variables

File: pandas/core/generic.py [hunk 2]
```diff
@@ -3589,13 +3589,20 @@ def _slice(self: FrameOrSeries, slobj: slice, axis=0) -> FrameOrSeries:
         result._set_is_copy(self, copy=is_copy)
         return result
 
-    def _iset_item(self, loc: int, value) -> None:
+    def _iset_item(self, loc: int, value, clear: bool_t = True) -> None:
```
Comment (jreback) [inline pandas/core/generic.py:3592]: where do we not clear?

Comment (jbrockmendel) [reply pandas/core/generic.py:3592]: we get here via NDFrame._maybe_cache_changed.

These are pretty spaghetti-ish, looking forward to cleaning them out

**Comment by jbrockmendel** [comment]
updated to remove the need for `clear` kwarg, so we end up not needing to touch DataFrame methods at all

**Comment by jreback** [comment]
thanks!

**Pull request merged**